### PR TITLE
fix backward read buffer

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

File: src/backward.rs [hunk 1]
```diff
@@ -39,16 +39,14 @@ pub unsafe fn json_rdb_load(rdb: *mut raw::RedisModuleIO) -> Value {
     match node_type {
         NodeType::Null => Value::Null,
         NodeType::Boolean => {
-            let mut str_len = 0;
-            let str_buffer = raw::load_string_buffer(rdb, &mut str_len);
-            Value::Bool(str_buffer.starts_with("1"))
+            let buffer = raw::load_string_buffer(rdb);
+            Value::Bool(buffer[0] == ('1' as u8))
```
Comment (gavrie) [inline src/backward.rs]: Instead of `('1' as u8)`, you can simply say `b'1'` (similar to Python).

File: src/backward.rs [hunk 2]
```diff
@@ -39,16 +39,14 @@ pub unsafe fn json_rdb_load(rdb: *mut raw::RedisModuleIO) -> Value {
...
         }
         NodeType::Integer => Value::Number(raw::load_signed(rdb).into()),
         NodeType::Number => Value::Number(Number::from_f64(raw::load_double(rdb)).unwrap()),
         NodeType::String => {
-            let mut str_len = 0;
-            let str_buffer = raw::load_string_buffer(rdb, &mut str_len);
-            Value::String(str_buffer.to_string())
+            let buffer = raw::load_string_buffer(rdb);
+            Value::String(buffer.to_string().unwrap())
```
Comment (gavrie) [inline src/backward.rs:49]: Are you OK with this panicking if the string is not UTF-8? It seems we should handle this more elegantly than with `unwrap()`.

**Review by gavrie — CHANGES_REQUESTED** [review]

File: src/backward.rs (hunk 2 above)
Comment (gkorland) [reply src/backward.rs:49]: yes, it should panic, it means we can't read the RDB and the server should shutdown with error

File: src/backward.rs (hunk 1 above)
Comment (gkorland) [reply src/backward.rs]: done

**Review by gavrie — DISMISSED** [review]

**Pull request merged**